packages/block-editor/src/components/block-binding-support/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import withBlockBindingSupport from './with-block-binding-support';
+
+export const BLOCK_BINDINGS_ALLOWED_BLOCKS = {
+	'core/paragraph': [ 'content' ],
+	'core/heading': [ 'content' ],
+	'core/image': [ 'url', 'title', 'alt' ],
+	'core/button': [ 'url', 'text', 'linkTarget' ],
+};
+
+export function isItPossibleToBindBlock( blockName ) {
+	return blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS;
+}
+
+export default function extendBlockWithBoundAttributes( settings, name ) {
+	if ( ! isItPossibleToBindBlock( name ) ) {
+		return settings;
+	}
+
+	return {
+		...settings,
+		/*
+		 * Expose relevant data through
+		 * the block context.
+		 */
+		usesContext: [
```

Comment:
This part will be handled by the outcome of [this pull request](https://github.com/WordPress/gutenberg/pull/58554) and, potentially, it won't be needed.

Reply:
Do you think it should be removed altogether from this PR? I think there's no harm in the current refactoring. We can remove in the future it if it's obsoleted by https://github.com/WordPress/gutenberg/pull/58554

Reply:
Oh, I see that https://github.com/WordPress/gutenberg/pull/58554 is removing it. So it doesn't matter either way probably 🙂 .

Reply:
Yes, we will remove that part of the filter. Whichever PR gets merged later would need to adapt to the changes.